src/metabase/events/sync_database.clj

```diff
@@ -1,49 +0,0 @@
-(ns metabase.events.sync-database
-  (:require
-   [clojure.core.async :as a]
-   [metabase.events :as events]
-   [metabase.models.database :refer [Database]]
-   [metabase.sync :as sync]
-   [metabase.sync.sync-metadata :as sync-metadata]
-   [metabase.util :as u]
-   [metabase.util.i18n :refer [trs]]
-   [metabase.util.log :as log]
-   [toucan2.core :as t2]))
-
-(def ^:private sync-database-topics
-  "The `Set` of event topics which are subscribed to for use in database syncing."
-  #{:database-create})
-
-(defonce ^:private ^{:doc "Channel for receiving event notifications we want to subscribe to for database sync events."}
-  sync-database-channel
-  (a/chan))
-
-
-;;; ------------------------------------------------ EVENT PROCESSING ------------------------------------------------
-
-
-(defn process-sync-database-event
-  "Handle processing for a single event notification received on the `sync-database-channel`"
-  [{topic :topic, object :item, :as event}]
-  ;; try/catch here to prevent individual topic processing exceptions from bubbling up.  better to handle them here.
-  (try
-    (when event
-      (when-let [database (t2/select-one Database :id (events/object->model-id topic object))]
-        ;; just kick off a sync on another thread
-        (future
-          (try
-            ;; only do the 'full' sync if this is a "full sync" database. Otherwise just do metadata sync only
-            (if (:is_full_sync database)
-              (sync/sync-database! database)
-              (sync-metadata/sync-db-metadata! database))
```

Comment:
Now that we removed `:database-trigger-sync` from `sync-database-topics` in [this PR](https://github.com/metabase/metabase/pull/31543) and `:is_full_sync` should be true by default for new databases (https://github.com/metabase/metabase/pull/31572), we can simply call `(sync/sync-database! database)` now.